### PR TITLE
DOC-164: Improve the documentation style guide

### PR DESCRIPTION
**Pull request by tylerkelly13**
Related Ticket: DOC-164

Description of Changes:
* Update and expand the existing documentation style guide.

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: configure/contributing-docs.md [hunk 1]
```diff
@@ -97,42 +100,234 @@ keywords:          // Space-separated list of keywords in the content.
 ---
 ```
 
-## Style Guide
+## Recommended tools
+
+The following tools are recommends for significant edits to the documentation. The warnings and notifications from these tools should only be used as _guides_, not as _rules_.
```
Comment (EkimChau) [inline configure/contributing-docs.md:105]: "recommends" -> "recommended"

**Review by EkimChau — CHANGES_REQUESTED** [review]

File: configure/contributing-docs.md (hunk 1 above)
Comment (tylerkelly13) [inline configure/contributing-docs.md:105]:
```suggestion
The following tools are recommended for significant edits to the documentation. The warnings and notifications from these tools should only be used as _guides_, not as _rules_.
```

Comment (tylerkelly13) [reply configure/contributing-docs.md]: thanks. fixed by: https://github.com/tinymce/tinymce-docs/pull/1741/commits/809a5dcbbae7595eca5e9952867b42baf9890074

File: _includes/misc/glossary.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
+<dl>
+  <dt>Accessible Rich Internet Applications (ARIA)</dt>
+  <dd>Acronym for <em>Accessible Rich Internet Applications</em> Spell out and describe on first mention within a
+    section.<br />
+    See: <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA">MDN Web Docs - ARIA</a>.</dd>
+  <dt>add-on</dt>
+  <dd>Do not use. Use “plugin” instead.</dd>
```
Comment (EkimChau) [inline _includes/misc/glossary.md:7]: do not use add-on?

**Review by EkimChau — APPROVED** [review]

File: _includes/misc/glossary.md (hunk 2 above)
Comment (tylerkelly13) [reply _includes/misc/glossary.md:7]: Consistency. We call them plugins everywhere, let's not confuse people.

Comment (lnewson) [reply _includes/misc/glossary.md:7]: I read this more of a suggestion, so something like this:
```suggestion
<dd>Do not use <em>add-on</em>. Use <em>“plugin”<em> instead.</dd>
```

Comment (tylerkelly13) [reply _includes/misc/glossary.md:7]: Damn. I thought I had replaced all the double quotes with em elements :(

File: _includes/misc/glossary.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
+<dl>
+  <dt>Accessible Rich Internet Applications (ARIA)</dt>
+  <dd>Acronym for <em>Accessible Rich Internet Applications</em> Spell out and describe on first mention within a
+    section.<br />
+    See: <a href="https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA">MDN Web Docs - ARIA</a>.</dd>
+  <dt>add-on</dt>
+  <dd>Do not use <em>add-on</em>. Use <em>“plugin”<em> instead.</dd>
```
Comment (tylerkelly13) [inline _includes/misc/glossary.md:7]:
```suggestion
  <dd>Do not use <em>add-on</em>. Use <em>“plugin”</em> instead.</dd>
```

**Review by lnewson — APPROVED** [review]

File: _includes/misc/glossary.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+  <dd>Do not use <em>add-on</em>. Use <em>“plugin”</em> instead.</dd>
+  <dt>anchor</dt>
+  <dd>Use when describing the Anchor plugin or the technical insertion of an anchor tag into HTML. When talking with a
+    business audience, use <em>bookmark</em> or add <em>bookmark</em> as a clarifier.</dd>
+  <dt>Application Programming Interface (API)</dt>
+  <dd>Use <em>API</em> in most circumstances. Spell it out on first mention only if the document is for a non-technical
+    audience.</dd>
+  <dt>asynchronous, async</dt>
+  <dd>Use <em>asynchronous</em> Do not use <em>async</em> </dd>
+  <dt>bookmark</dt>
```
Comment (tylerkelly13) [inline _includes/misc/glossary.md:16]:
```suggestion
  <dt>back end, back-end, backend</dt>
  <dd>Do not use backend. Use <em>back end</em> as a noun. Use back-end as a compound adjective.</dd>
  <dt>bookmark</dt>
```

**Pull request merged**